hackathon_site/dashboard/frontend/src/components/general/ProductOverview/ProductOverview.test.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import SideSheetRight from "../SideSheetRight/SideSheetRight";
+import { ProductOverview, EnhancedCartForm } from "./ProductOverview";
+import {
+    render,
+    fireEvent,
+    waitForElementToBeRemoved,
+    waitFor,
+    waitForElement,
+} from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+describe("Product Overview", () => {
+    const testDetail = {
+        name: "Arduino",
+        type: "red",
+        total: 30,
+        available: 29,
+        img: "https://www.filterforge.com/more/help/images/size200.jpg",
+        category: ["MCU", "FPGA"],
+        manufacturer: "Canakit",
+        model_num: "Model 3B+",
+        datasheet: "link",
+        notes: ["- For micropython ask for image", "- randomnerdtutorials.com"],
+        constraints: ["- Max 1 of this item", "- Max 3 microcontroller labelled red"],
+        quantity: 3,
+    };
+
+    test("render", () => {
+        const addCartMock = jest.fn();
+
+        const { asFragment } = render(
+            <SideSheetRight title="Product Overview" isVisible={true}>
+                <ProductOverview detail={testDetail} addCartFunction={addCartMock} />
+            </SideSheetRight>
+        );
+
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    test("add to cart button calls the correct function", () => {
+        const addCartMock = jest.fn();
+
+        const { getByText, getByRole } = render(
+            <SideSheetRight title="Product Overview" isVisible={true}>
```

Comment:
Do you actually need to have `SideSheetRight` as part of these tests? You should be able to test a lof of the components within `ProductOverview` (especially the form) individually